addons/sale_timesheet/models/project.py

```diff
@@ -47,13 +49,43 @@ def _default_timesheet_product_id(self):
         default=_default_timesheet_product_id)
     warning_employee_rate = fields.Boolean(compute='_compute_warning_employee_rate')
 
-    @api.depends('allow_billable', 'sale_order_id', 'partner_id', 'pricing_type')
+    @api.depends('sale_line_id', 'sale_line_employee_ids', 'allow_billable')
+    def _compute_pricing_type(self):
+        billable_projects = self.filtered('allow_billable')
+        for project in billable_projects:
+            if project.sale_line_employee_ids:
+                project.pricing_type = 'employee_rate'
+            elif project.sale_line_id:
+                project.pricing_type = 'fixed_rate'
+            else:
+                project.pricing_type = 'task_rate'
+        (self - billable_projects).update({'pricing_type': False})
+
+    def _search_pricing_type(self, operator, value):
+        if operator not in ('=', '!=') or not ((isinstance(value, bool) and value is False) or isinstance(value, str)):
+            raise UserError(_('Operation not supported'))
+        if value is False:
+            return [('allow_billable', operator, value)]
+        if operator == '=':
+            if value == 'task_rate':
+                return [('sale_line_employee_ids', '=', False), ('sale_line_id', '=', False)]
+            if value == 'fixed_rate':
+                return [('sale_line_employee_ids', '=', False), ('sale_line_id', '!=', False)]
+            if value == 'employee_rate':
+                return [('sale_line_employee_ids', '!=', False), ('sale_line_id', '!=', False)]
+        else:
+            if value == 'task_rate':
+                return ['|', ('sale_line_employee_ids', '!=', False), ('sale_line_id', '!=', False)]
+            if value == 'fixed_rate':
+                return ['|', ('sale_line_employee_ids', '!=', False), ('sale_line_id', '=', False)]
+            if value == 'employee_rate':
+                return [('sale_line_employee_ids', '=', False)]
```

Comment:
Couldn't we store the different domains in variables and use `expression.distribute_not(['!'] + domain_xxx)` in the else in order to ease maintenance? Just an idea I have at the time I write... Not sure it is really nice 🙂

Reply:
That's a good question, I didn't know it, I will test it, thanks ! :)